desktop/appcontainer/Banner.scss

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2021 Extremely Heavy Industries Inc.
+ */
+body.xh-app .xh-banner {
+  height: var(--xh-banner-height-px);
+
+  & > .xh-icon {
+    margin-right: var(--xh-pad-half-px);
+  }
+
+  &__message {
+    margin-right: var(--xh-pad-px);
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
```

Comment:
I tested with a very long message (pretty easy to imagine use case) and the message div pushes the action and dismiss buttons off the edge - we should resolve that if you have a chance to look.

Might be easier to do if the action button also floats to the right - like on mobile - so the message div gets the flex but no more.  I was wondering if that would look better anyway, so the action button is in a more consistent place. (Realize existing styling comes from update bar.). 

I'm happy either way, but I do think we should ensure it clips so those buttons remain accessible at all screen widths.

Reply:
Floating the action button to the right is a great idea, had the same thought myself. Then the message will take the flex and elide on overflow.